server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -719,6 +719,12 @@ public IndexMetadata getIndexSafe(Index index) {
             .orElse(Collections.emptyMap());
     }
 
+    public Map<String, SingleNodeShutdownMetadata> nodeShutdowns() {
```

Comment:
No, there's a couple things that prevent this:
1) In `TransportPutShutdownNodeAction` when we get a `PUT` for a node that already has a record, it's updated rather than added to, and
2) The data structure used to store the `SingleNodeShutdownMetadata` (the `Map` in the line you're commenting on) is keyed by node UUID, so it should be impossible to have multiple records for the same key.

Since the node id is duplicated in the `SingleNodeShutdownMetadata` as well, it's conceivable that in the case of a bug we could end up with a mismatch between the id used for keying and the id used in the object, but I don't think that's likely.

Reply:
Yeah, sorry if this was unclear, what I meant was whether it could be a reasonable use case to have both a RESTART and one of the two others at the same time. Not that there is anything wrong in this PR or anything, more wanted to bring this to your attention for possible discussion (and maybe you discussed it already and discarded the use case?).